gulp/tasks/build.js

```diff
@@ -144,21 +128,23 @@ function compileIndvES2015ModulesToBrowser() {
     },
     module: {
       rules: [{
-        test: /\.ts(x?)$/,
-        exclude: /node_modules/,
-        use: [
-          babelLoader,
-          tsLoader
-        ]
-      }, {
-        test: /\.js$/,
+        test: /\.tsx?$/,
         exclude: /node_modules/,
-        use: [
-          babelLoader
-        ]
+        loader: 'awesome-typescript-loader',
+        options: {
+          useCache: true,
+          useBabel: true,
+          babelOptions: config.babelWebpack
+        }
       }]
     },
     plugins: [
+      new CheckerPlugin(),
+      new webpack.DefinePlugin({
```

Comment:
I diff'ed doing the compression w/ and w/o this. For me I was seeing a little bit better results compiling database without this.

### Size Comparison

| File | With `webpack.DefinePlugin` | Without `webpack.DefinePlugin`
|----------|-------------|------|
| firebase-database.js | 166.19 KB | 166.16 KB |